pkg/internal/token/msalcacheprovider.go

```diff
@@ -0,0 +1,81 @@
+package token
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/cache"
+)
+
+// MSALCacheProvider exports and replaces in-memory cache data. It doesn't support nil Context or
+// define the outcome of passing one. A Context without a timeout must receive a default timeout
+// specified by the implementor. Retries must be implemented inside the implementation.
+type MSALCacheProvider interface {
```

Comment:
why duplicate from https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/v1.4.2/apps/cache/cache.go#L47